### PR TITLE
[#3850] Fix the problem when setting content-type by ResponseEntity.header() does not take effect

### PR DESCRIPTION
**Pull request by Sayaka617**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/DownloadUtils.java [hunk 1]
```diff
@@ -44,7 +44,11 @@ public static void prepareDownloadHeader(HttpServletResponseEx responseEx, Part
       return;
     }
     if (responseEx.getHeader(HttpHeaders.CONTENT_TYPE.toString()) == null) {
-      responseEx.setHeader(HttpHeaders.CONTENT_TYPE.toString(), part.getContentType());
+      if (responseEx.getContentType() != null) {
+        responseEx.setHeader(HttpHeaders.CONTENT_TYPE.toString(), responseEx.getContentType());
+      } else {
+        responseEx.setHeader(HttpHeaders.CONTENT_TYPE.toString(), part.getContentType());
+      }
```
Comment (liubao68) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/DownloadUtils.java:51]: Do you find this problem in spring boot embedded tomcat or standalone tomcat? And would you mind to add a test case in `demo` folders (there are many test cases and you can run each demo from your IDE).

Comment (Sayaka617) [reply foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/DownloadUtils.java:51]: it's in spring boot embedded tomcat .
Of course , I'll add some test cases .

**Comment by liubao68** [comment]
@Sayaka617 Do not use force-push if possible, so that we can now what's changed after last review. Do don't worry about the commits, we can use squash merge when merge the PR.   
I triggered the CI again as this is your first PR.

**Comment by Sayaka617** [comment]
> @Sayaka617 Do not use force-push if possible, so that we can see what's changed after last review. Do don't worry about the commits, we can use squash merge when merge the PR. I triggered the CI again as this is your first PR.

okay，i got it . I'll avoid this next time . thank you

**Comment by liubao68** [comment]
@Sayaka617  This PR is OK now.  Do you mind to cherry-pick this PR to 2.8.x branch?

**Pull request merged**